rescan.go

```diff
@@ -600,12 +591,31 @@ func rescan(chain ChainSource, options ...RescanOption) error {
 	// handleBlockDisconnected is a helper closure that handles a new block
 	// disconnected notification.
 	handleBlockDisconnected := func(ntfn *blockntfns.Disconnected) error {
-		log.Debugf("Rescan disconnect block %d (%s)\n", curStamp.Height,
+		log.Debugf("Rescan disconnect block %d (%s)", curStamp.Height,
 			curStamp.Hash)
 
+		blockDisconnected := ntfn.Header()
+
+		// Check whether the block being disconnected is one for which
+		// we've queued up to retry. If it is, we'll remove it and any
+		// others that follow as they are now considered stale.
+		retryIdx := -1
+		for i, retryBlock := range retryBlocks {
+			retryHeader := retryBlock.Header()
+			if retryHeader.BlockHash() == blockDisconnected.BlockHash() {
+				retryIdx = i
+				break
+			}
+		}
```

Comment:
Looking at the amount of code that manipulates `retryBlocks`, it may be justified to create a separate struct for it with methods to manipulate the list.

Reply:
Done!